CalibCalorimetry/HcalAlgos/src/HcalSiPMnonlinearity.cc

```diff
@@ -0,0 +1,19 @@
+#include <iostream>
+#include "Math/Polynomial.h"
+#include "CalibCalorimetry/HcalAlgos/interface/HcalSiPMnonlinearity.h"
+
+// Assume parameters come to us from the reco side; i.e.,
+// true pes = corfun(pixelsfired). But we want to invert that.
+//
+int HcalSiPMnonlinearity::getPixelsFired(int inpes) const
+{
+  ROOT::Math::Polynomial p(a2,b1,c0,-inpes);
+  std::vector<double> roots = p.FindRealRoots();
```

Comment:
Yes, right now this is just left in place for future studies.

The ROOT functions do a fair amount of useful processing (it's not just a direct clone of gsl), so I would prefer to leave it this way.